src/object/cli_obj.c

```diff
@@ -2857,6 +2857,7 @@ obj_req_fanout(struct dc_object *obj, struct obj_auxi_args *obj_auxi,
 			tgt += req_tgts->ort_grp_size;
 		else
 			tgt++;
+		obj_auxi->new_shard_tasks = 1;
```

Comment:
this looks incorrect, and cause the CI test failures.
new_shard_tasks' usage is to indicate when io retry should re-create new shard tasks, or can reuse original shard tasks. Common usage is to remove old shard tasks and set that flag.
I checked some related code, and found there actually is a problem that forgot to remove old shard task, that may cause mem-leak. But that only happens in the rare EC FETCH_AGAIN case, I pushed https://github.com/daos-stack/daos/pull/6669 to fix that case just now.

Reply:
If IO fail before the shard task is created, and then retry, even even the new shard task is created, these shard task won't be scheduled. see obj_shard_task_schedule().

So new_shard_tasks can be used to indicate there is new shard tasks are created, no matter for retry task or normal task?

Reply:
Hmm, probably adding a flag would be better here. I added shards_scheded flag in obj_auxi, please check. Thanks.

Reply:
ok, I just tested it works now. We probably can try to reduce those flags later, but fine for now. thanks